dev-docs/RFCs/v4.0.0/terms_for_concepts_and_datatypes.md

```diff
@@ -0,0 +1,60 @@
+# RFC: Terms for H3 Concepts and Data Types
+
+- **Authors**: AJ Friend
+- **Date**: 2020-02-02
+- **Status**: Work In Progress
+
+## Motivation
+
+Concepts like `hexagon`, `cell`, and `index` are currently used ambiguously
+in the H3 codebase and in documentation. This can cause confusion, for example,
+when a function might only work on strict H3 *hexagons*, but fails when
+encountering H3 *pentagons*.
+
+Reaching a consensus on the precise, technical language used when discussing H3 concepts will clarify future library discussions and improve end-user documentation.
+
+## Proposals
+
+## Concepts
+
+The following technical terms should be used in the documentation, the H3 codebase, and precise technical discussions of the library.
+
+- `H3Index`, "H3 index", or "index"
+    - an unsigned 64-bit integer representing **any** H3 object (hexagon, pengaton, unidirectional edge, ...)
```

Comment:
Nit: `pentagon`

Reply:
More substantively - we currently refer to this in function names as `h3`, e.g. `h3ToGeo` etc. It would be good to have an opinion on whether we should keep this convention in function names, or whether this RFC would suggest updating to `cellToGeo`, etc.

Reply:
Oh, I'm all for doing something like `cellToGeo`!

I was thinking of doing the function renaming (`k_ring`, etc.) in a separate RFC, but I think there's enough overlap that I can just turn this into an all-encompassing "naming" RFC.